### DISCUSSION TITLE
ContentPart with ContentFields: precompile fields by code

### DISCUSSION THREAD

**Discussion by PiemP** [OrchardCMS/OrchardCore#13942] [Q&A]
Good evening everyone, 

I'm trying to precompile some fields of my ContentPart by take some data from a database. The ContentPart is created with a migration and have some ContentFields.

I have tried  with a `DisplayDriver` but I don´t know how render the shape of each ContentFields. I have tried with a `ContentPartHandler` but I don`t know how set the values in the ContentFields and return them to the view. 

Anyone have experience about it? Have I to raise white flag and implement a shape for this ContentPart from zero?

Thank you.

**Comment by jtkech** [comment]
You also need to define at least one ContentType that will have this ContentPart, this through the admin UI or a migration. Then this is when creating/editing an item of such a content type that the default field editor shapes will be rendered.

**Reply by PiemP** [reply]
Thank you @jtkech, 

your solution works. I believed that I have understand how to pass data to the edit view, but for now I'm struggling to understand this step. 😅

**Reply by PiemP** [reply]
I don't have understand how pass data from a `DisplayDriver` to the edit view. With the `TextField` works in some way (I presume is not the right way) but with the `BooleanField` and `NumericField` I can't set the value before render the edit view.
